libusb/os/darwin_usb.c

```diff
@@ -1568,6 +1568,7 @@ static int darwin_set_interface_altsetting(struct libusb_device_handle *dev_hand
 
   kresult = (*(cInterface->interface))->SetAlternateInterface (cInterface->interface, altsetting);
   if (kresult != kIOReturnSuccess)
+    usbi_warn (HANDLE_CTX (dev_handle), "SetAlternateInterface: %s", darwin_error_str(kresult));
     darwin_reset_device (dev_handle);
```

Comment:
Missing braces so only the warning is conditional, while the reset is always done.

Reply:
Bugger. I clearly have been exposed to Python recently :),.Thanks for the catch!

Reply:
Fixed in 6d7069f